python/lsst/meas/base/forcedPhotImage.py

```diff
@@ -111,8 +163,30 @@ class ForcedPhotImageTask(lsst.pipe.base.CmdLineTask):
     ConfigClass = ForcedPhotImageConfig
     _DefaultName = "processImageForcedTask"
 
-    def __init__(self, butler=None, refSchema=None, **kwds):
-        super(lsst.pipe.base.CmdLineTask, self).__init__(**kwds)
+    def getInitOutputDatasets(self):
+        return {"outputSchema": afwTable.SourceCatalog(self.measurement.schema)}
+
+    def adaptArgsAndRun(self, inputData, inputDataIds, outputDataIds, butler):
+        if 'measCat' not in inputData.keys():
```

Comment:
There's some code duplication in this block. It looks identical but with "TractPatch" instead of "VisitDetector"

Reply:
Moved into new method.